README.md

```diff
@@ -1,121 +1,131 @@
+  description: "A schema with two version markers is invalid",
+  
+  // The actual schema, embedded in a s-expression. Test runners must convert this s-expression to a document.
+  invalid_schema:(
+    $ion_schema_2_0
+    $ion_schema_1_0
+    schema_header::{}
+    type::{
+      name: foo,
+      type: int,
+    }
+    schema_footer::{}
+  )
+}
+
+// A parameterized test case that checks that multiple type definitions are correctly recognized as invalid
+// Test runner implementations should append the zero-based index of the test case to the test description. 
+// E.g. "Regex must be a string [1]"
+$test::{
+  // A useful description of the test to aid with debugging, understanding the spec, etc.
+  description: "Regex must be a string",
+  
+  // A list of types that are invalid. They should be detected as invalid by both the Ion Schema implementation and
+  // by the types defined in ion-schema-schemas.
+  invalid_types: [
+    { regex: 1 },
+    { regex: false },
+    { regex: [] },
+    { regex: () },
+  ]
+}
+```
 
-For example:
-> ```
-> invalid_type::{ type: 5 }           // '5' is not a valid type reference
-> ```
-
-### Invalid schema definition test
-To assert that a schema definition should be recognized as invalid, the `invalid_schema` annotation is used.
-
-For example:
-> ```
-> invalid_schema::document::'''
->   schema_header::{}
->   type::{ type: unknown_type }      // this type reference is invalid
->   schema_footer::{}
-> '''
-> ```
-
-### Validation details test
-To assert that the Violations object returned by `Type.validate()` includes the expected details, a type or schema definition is provided,
-followed by a struct with the `test_validation` annotation.
-
-For example:
-> ```
-> type::{
->   fields: {
->     age: int,
->   },
->   type: $any,
-> }
-> test_validation::{
->   value: {age: five},       // the value to test
-> 
->   // alternatively, mutliple test values may be specified:
->   // values: [{age: five}, {age: null}],
-> 
->   // in the context of a schema, the type to validate against must be specified, e.g.:
->   // type: X,
-> 
->   violations: [             // corresponds to a Violations object returned by Type.validate()
->     {
->       constraint: { fields: { age: int } },
->       code: fields_mismatch,
->       children: [
->         {
->           fieldName: "age",
->           value: five,
->           violations: [ { constraint: { type: int }, code: type_mismatch } ],
->         },
->       ],
->     },
->   ],
-> }
-> ```
+General notes:
+* The directory and file structure exists primarily for organizational purposes and is not relevant to successful execution of the tests (except for tests within the schema directory that rely on schema import functionality)
+* Every test file should be a valid Ion Schema document (though it may be an empty schema if there are no types defined).
+* Documents are represented as annotated _s-expressions_ for `should_match` and `should_not_match`.
```

Comment:
(Question) Why have we changed the document types defined in the tests to use sexpression?

Reply:
By embedding them inside a s-expression...
1. Your IDE can tell you if you have written some malformed Ion in a s-expression. It will not do that for Ion inside a string.
2. If the embedded document is not valid Ion (valid in the sense that it conforms to the Ion specification), the test runner can fail-fast for that test file having malformed Ion rather than failing late because an exception is thrown when we try to run that particular test case.

I am aware that the S-expression-wrapped document is no longer a separate Ion stream (so it cannot have a separate symbol table from the document that it is embedded in, etc.)—however, I can't think of any reason why it would matter that it's now part of the same Ion stream rather than being a separate Ion stream, since Ion Schema operates on the data model rather than at an encoding level.